rust/metaplex/program/tests/create_store.rs

```diff
@@ -0,0 +1,67 @@
+#![cfg(feature = "test-bpf")]
+
+use metaplex::state::{CONFIG, PREFIX, Store, StoreConfig, Key};
+use metaplex::{instruction, id};
+use num_traits::FromPrimitive;
+use solana_program::{config, pubkey::Pubkey};
+use solana_program_test::*;
+use solana_sdk::signature::Signer;
+use solana_program::borsh::try_from_slice_unchecked;
+use solana_sdk::{
+    instruction::InstructionError,
+    signer::keypair::Keypair,
+    transaction::{Transaction, TransactionError},
+    transport::TransportError,
+};
+mod create_store {
+    use metaplex::state::StoreConfig;
+
+    use super::*;
+
+    pub fn program_test<'a>() -> ProgramTest {
+        ProgramTest::new("metaplex", metaplex::id(), None)
+    }
+
+    #[tokio::test]
+    async fn set_store_success() {
+        let mut context = program_test().start_with_context().await;
+        let payer_key = context.payer.pubkey();
+        let mid = id();
+        let store_seeds = &[
+            PREFIX.as_bytes(),
+            mid.as_ref(),
+            payer_key.as_ref(),
+        ];
+        let (store_key, _) = Pubkey::find_program_address(store_seeds, &mid);
+        let config_seeds = &[
+            PREFIX.as_bytes(),
+            mid.as_ref(),
+            CONFIG.as_bytes(),
+            store_key.as_ref(),
+        ];
+        let (config_key, _) = Pubkey::find_program_address(config_seeds, &mid);
+        let tx = Transaction::new_signed_with_payer(
+            &[instruction::create_set_store_v2_instruction(
+                mid,
+                store_key,
+                config_key,
+                payer_key,
+                payer_key,
+                true,
+                Some("https://notgoogle.com".to_string()),
+            )],
+            Some(&payer_key),
+            &[&context.payer],
+            context.last_blockhash,
+        );
+        context.banks_client.process_transaction(tx).await;
+        let store = context.banks_client.get_account(store_key).await.unwrap().unwrap();
+        let store_data: Store = try_from_slice_unchecked(&store.data).unwrap();
+        let config = context.banks_client.get_account(config_key).await.unwrap().unwrap();
+        let config_data: StoreConfig = try_from_slice_unchecked(&config.data).unwrap();
+       
+        assert_eq!(store_data.key, Key::StoreV1);
+        assert_eq!(config_data.key, Key::StoreConfigV1);
+        assert_eq!(config_data.settings_uri, Some("https://notgoogle.com".to_string()));
+    }
+}
```

Comment:
Ideally we'd add some more test here to check the _not happy_ paths.
In particular we should check for known attack vectors and at the very least cover the case where the `store_config` isn't empty but not owned by the program.

Reply:
Agreed, however you're adding a new contract and we don't just need to make sure that it works, but also that it is secure and handles incorrect inputs correctly.
That is my opinion at least.